plan/typeinferer.go

```diff
@@ -319,6 +319,12 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
 	case "connection_id":
 		tp = types.NewFieldType(mysql.TypeLonglong)
 		tp.Flag |= mysql.UnsignedFlag
+	case "find_in_set":
+		if x.Args[0].GetType().Tp == mysql.TypeNull || x.Args[1].GetType().Tp == mysql.TypeNull {
```

Comment:
find_in_set always has return value of BIGINT. You can check it out using mysqlworkbench.